src/PowerShell/PowerShellManager.cs

```diff
@@ -48,79 +58,78 @@ internal PowerShellManager(ILogger logger)
             _pwsh.Streams.Warning.DataAdding += streamHandler.WarningDataAdding;
         }
 
-        internal void AuthenticateToAzure()
+        internal void InitializeRunspace()
         {
-            // Check if Az.Profile is available
-            Collection<PSModuleInfo> azProfile = _pwsh.AddCommand("Get-Module")
-                .AddParameter("ListAvailable")
-                .AddParameter("Name", "Az.Profile")
-                .InvokeAndClearCommands<PSModuleInfo>();
+            // Add HttpResponseContext namespace so users can reference
+            // HttpResponseContext without needing to specify the full namespace
+            _pwsh.AddScript($"using namespace {typeof(HttpResponseContext).Namespace}").InvokeAndClearCommands();
 
-            if (azProfile.Count == 0)
+            // Set the PSModulePath
+            Environment.SetEnvironmentVariable("PSModulePath", Path.Join(AppDomain.CurrentDomain.BaseDirectory, "Modules"));
+        }
+
+        internal void InvokeProfile()
+        {
+            IEnumerable<string> profiles = Directory.EnumerateFiles(FunctionAppRootLocation, PROFILE_FILENAME, ENUMERATION_OPTIONS);
+            if (profiles.Count() == 0)
             {
-                _logger.Log(LogLevel.Trace, "Required module to automatically authenticate with Azure `Az.Profile` was not found in the PSModulePath.");
+                _logger.Log(LogLevel.Trace, $"No 'Profile.ps1' found at: {FunctionAppRootLocation}");
                 return;
             }
 
-            // Try to authenticate to Azure using MSI
-            string msiSecret = Environment.GetEnvironmentVariable("MSI_SECRET");
-            string msiEndpoint = Environment.GetEnvironmentVariable("MSI_ENDPOINT");
-            string accountId = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
-
-            if (!string.IsNullOrEmpty(msiSecret) &&
-                !string.IsNullOrEmpty(msiEndpoint) &&
-                !string.IsNullOrEmpty(accountId))
+            var dotSourced = new StringBuilder(". ").Append(QuoteEscapeString(profiles.First()));
```

Comment:
Instead of using `. <path-to-profile.ps1>`, let's do this magic: `Import-Module <profile-file-path> -PassThru | Remove-Module`.
`Import-Module` on a `.ps1` file will evaluate the script in the global scope.
This causes problem to us for supporting `entrypoint` in `.ps1` script, but helps us in the `profile.ps1` case. By doing this, we  no longer need to escape the file path.

Reply:
That's some magic! changed